cypress/integration/sidebar.spec.js/sidebar.spec.js

```diff
@@ -0,0 +1,95 @@
+/// <reference types="cypress" />
+
+describe('sidebar', () => {
+  // Based on sidebar.js
+  const defaultSectionName = 'getting-started';
+  const basepath = '/docs/2.0/'
+  beforeEach(() => {
+    // Takes to starting url which is baseURL
+    cy.visit(`${basepath}`)
+  })
+
+  describe('Full screen', () => {
+    beforeEach(() => {
+      cy.viewport(1280, 720)
+    })
+
+    it('should load sidebar in full screen', () => {
+      // A nav should exist and one should have class sidebar
+      cy.get('nav').should('exist').should('have.class', 'sidebar')
+
+      // Sidebar should contain at least the default Section
+      cy.get(`[data-section="${defaultSectionName}"]`).should('exist')
+      cy.get('nav.sidebar').children('.sidebar-item-group').children('ul').should('have.length.greaterThan', 0)
+    })
+
+    it('should close getting started and open pipelines', () => {
+      // Check defaultSectionName open then close defaultSectionName
+      cy.get('nav.sidebar').children('.sidebar-item-group').children('ul').children(`[data-section="${defaultSectionName}"]`).should('not.have.class', 'closed').children('.list-wrap').click()
+      // Check defaultSectionName closed then open
+      cy.get('nav.sidebar').children('.sidebar-item-group').children('ul').children(`[data-section="${defaultSectionName}"]`).should('have.class', 'closed').click().should('not.have.class', 'closed')
+    })
+
+    it('should bring you to right url', () =>{
+      // Click second element in getting started which is migration-intro
+      cy.get('nav.sidebar').children('.sidebar-item-group').children('ul').children(`[data-section="${defaultSectionName}"]`).children().next().click()
+      cy.url().should('include', 'migration-intro')
```

Comment:
do we want to hardcode stuff like this? is there anyway this can be done in a less hardcoded fashion?

Reply:
can we derive all the URLs in this file from https://github.com/circleci/circleci-docs/blob/ea8c1f7c6409952ec35b7e2b397a89b3691a0f07/jekyll/_data/sidenav.yml ?

Reply:
https://docs.cypress.io/api/commands/children#Selector 
There is a lot of chaining that takes place unless we place id's on things. Then we can just get the id we want no matter where it is in the DOM

Reply:
i don't have a problem with the ids/classes. its the URLs.

but in the future if a technical writer wants to modify some links, I'm not sure if they're comfortable fishing around in here to modify this as well. Any simple link change would become a 2 step process.

Reply:
actually i just reviewed the logic here @chantelle11 

https://github.com/circleci/circleci-docs/blob/5f29a7f3aaab04db1ffafdc03b950ad67565947e/jekyll/_includes/sidebar.html

all the data-sections are coming from `sidenav.yaml`. I think the correct way to do this is rely on this Single Source of Truth.

In fact, since you have a SSOT, I think you can basically do a for loop over `sidenav.yaml` and create a bunch of cypress tests for everything in YAML file.

Reply:
+1 to have the `sidenav.yml` being the source of truth. We have been waiting on cypress for a while now so if it is something that requires extensive work, let's create a ticket for the next sprint so we can ship this ASAP.

Reply:
I think what I need to do is take the yml make it into a json object, parse the json object and then use the object as the source of truth. im just trying to figure out how to parse the yml still